frontend/src/toolbar/flags/flagsToolbarLogic.ts

```diff
@@ -50,6 +50,8 @@ export const flagsToolbarLogic = kea<flagsToolbarLogicType>([
                 getUserFlags: async (_, breakpoint) => {
                     const params = {
                         groups: getGroups(values.posthog),
+                        // Pass app user's distinct_id so flags are evaluated for the correct user context
+                        distinct_id: values.posthog?.get_distinct_id(),
```

Comment:
discussed this [offline](https://posthog.slack.com/archives/C07Q2U4BH4L/p1755195334799559?thread_ts=1755191207.268209&cid=C07Q2U4BH4L), but TL;DR this shouldn't change anything because we already have access to a distinct_id of the user associated with this request from the `temporary_token` that gets appending to the toolbar API calls.